zebra-consensus/src/transaction/tests.rs

```diff
@@ -451,6 +454,116 @@ async fn v5_transaction_with_transparent_transfer_is_rejected_by_the_script() {
     );
 }
 
+/// Test if signed V4 transaction with a dummy [`sprout::JoinSplit`] is accepted.
+///
+/// This test verifies if the transaction verifier correctly accepts a signed transaction.
+#[tokio::test]
+async fn v4_with_signed_sprout_transfer_is_accepted() {
+    let network = Network::Mainnet;
+    let network_upgrade = NetworkUpgrade::Canopy;
+
+    let canopy_activation_height = network_upgrade
+        .activation_height(network)
+        .expect("Canopy activation height is not set");
+
+    let transaction_block_height =
+        (canopy_activation_height + 10).expect("Canopy activation height is too large");
+
+    // Create a fake Sprout join split
+    let (joinsplit_data, signing_key) = mock_sprout_join_split_data();
+
+    let mut transaction = Transaction::V4 {
+        inputs: vec![],
+        outputs: vec![],
+        lock_time: LockTime::Height(block::Height(0)),
+        expiry_height: (transaction_block_height + 1).expect("expiry height is too large"),
+        joinsplit_data: Some(joinsplit_data),
+        sapling_shielded_data: None,
+    };
+
+    // Sign the transaction
+    let sighash = transaction.sighash(network_upgrade, HashType::ALL, None);
```

Comment:
One thing I'm worried about is that this test _assumes_ that `Transaction::sighash` works correctly. I think it's probably a reasonable assumption, but maybe there should be a specific test for `Transaction::sighash` as well?

Reply:
This feels like a question for @conradoplg ?

The underlying sighash implementation has tests. And we usually don't bother testing simple, obviously correct wrapper methods.

This ticket might also have some context:
Also use librustzcash for signature hashes (sighash) in V4 transactions and earlier (#2183)